lib/elixir/lib/config/provider.ex

```diff
@@ -0,0 +1,203 @@
+defmodule Config.Provider do
+  @moduledoc """
+  Specifies a provider API that loads configuration during boot.
+
+  Config providers are typically used during releases to load
+  external configuration while the system boots. This is done
+  by starting the VM with the mininum amount of applications
+  running, then invoking all of the providers, and then
+  restarting the system. This requires a mutable configuration
+  file in disk, as the result of the providers are written to
+  disk. See `mix release` for more information and examples.
+  """
+
+  @type config :: keyword
+  @type state :: term
+
+  @typedoc """
+  A path pointing to a configuration file.
+
+  Since configuration files are often accessed on target machines,
+  it can be expressed either as:
+
+    * a binary representing an absolute path
+
+    * a tuple {:system, system_var, path} where the config is the
```

Comment:
could `{:system, "RELEASE_SYS_CONFIG", ".config"}` be replaced by a `:init` (or similar) that would be translated into reading the config path from `:init.get_argument/1`? Or the ENV var has other use cases?

Reply:
That's an interesting idea but the user could pass an extra `-config` and then which one we pick could be a bit too confused. We could aim for the sys.config one but I would prefer to keep those decoupled.